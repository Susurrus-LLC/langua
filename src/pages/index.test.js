import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { PureHome as Home } from '.'

describe('Frequen', () => {
  const data = {
    site: {
      siteMetadata: {
        title: 'Langua',
        toolInfo: {
          gen: {
            title: 'Gen',
            link: 'gen',
            description: 'Gen description'
          }
        },
        toolInfo: {
          frequen: {
            title: 'Frequen',
            link: 'frequen',
            description: 'Frequen description'
          }
        },
        toolInfo: {
          deriv: {
            title: 'Deriv',
            link: 'deriv',
            description: 'deriv description'
          }
        },
        toolInfo: {
          lex: {
            title: 'Lex',
            link: 'lex',
            description: 'Lex description'
          }
        },
        toolInfo: {
          morph: {
            title: 'Morph',
            link: 'morph',
            description: 'Morph description'
          }
        },
        toolInfo: {
          tree: {
            title: 'Tree',
            link: 'tree',
            description: 'Tree description'
          }
        }
      }
    }
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Home data={data} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately', () => {
    let tree = renderer.create(<Home data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
