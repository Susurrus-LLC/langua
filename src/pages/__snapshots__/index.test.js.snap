// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frequen renders accurately 1`] = `
<div
  className="app"
>
  <header
    className="header"
  >
    <div
      className="titleArea"
    >
      <h1
        className="siteTitle"
      >
        <a
          className="titleLink"
          href="/"
        />
      </h1>
      <p
        className="subtitle"
      />
    </div>
    <nav
      className="nav"
    >
      <ul
        className="menu"
      >
        <li
          className="navLink"
        >
          <a
            href="/gen"
          >
            Gen
          </a>
        </li>
        <li
          className="navLink disabled"
        >
          <a
            href="/morph"
          >
            Morph
          </a>
        </li>
        <li
          className="navLink"
        >
          <a
            href="/deriv"
          >
            Deriv
          </a>
        </li>
        <li
          className="navLink"
        >
          <a
            href="/frequen"
          >
            Frequen
          </a>
        </li>
        <li
          className="navLink disabled"
        >
          <a
            href="/tree"
          >
            Tree
          </a>
        </li>
        <li
          className="navLink disabled"
        >
          <a
            href="/lex"
          >
            Lex
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    className="main"
  >
    <p
      className="siteDescription"
    >
      Langua is a suite of language tools that can be used for testing, experimentation, learning, analysis, or creation.
    </p>
    <div
      className="grid"
    >
      <a
        className="panel"
        href="undefined/"
      >
        <h2
          className="toolTitle"
        />
        <p
          className="toolDescription"
        >
          Languaundefined undefined
        </p>
      </a>
      <a
        className="panel disabled"
        href="undefined/"
      >
        <h2
          className="toolTitle"
        />
        <p
          className="toolDescription"
        >
          Languaundefined undefined
        </p>
      </a>
      <a
        className="panel"
        href="undefined/"
      >
        <h2
          className="toolTitle"
        />
        <p
          className="toolDescription"
        >
          Languaundefined undefined
        </p>
      </a>
      <a
        className="panel"
        href="undefined/"
      >
        <h2
          className="toolTitle"
        />
        <p
          className="toolDescription"
        >
          Languaundefined undefined
        </p>
      </a>
      <a
        className="panel disabled"
        href="tree/"
      >
        <h2
          className="toolTitle"
        >
          Tree
        </h2>
        <p
          className="toolDescription"
        >
          LanguaTree Tree description
        </p>
      </a>
      <a
        className="panel disabled"
        href="undefined/"
      >
        <h2
          className="toolTitle"
        />
        <p
          className="toolDescription"
        >
          Languaundefined undefined
        </p>
      </a>
    </div>
    <p
      className="siteDescription"
    >
      Langua
       is a full Progressive Web App; once the site has been loaded, its full functionality is available even offline. 
      Langua
       
      can also be added to the homescreen of your tablet or mobile device for easy access.
    </p>
    <div
      className="Toastify"
    />
  </main>
  <footer
    className="footer"
  >
    <p
      className="copyright"
    >
      <a
        href="undefined/blob/master/CHANGELOG.md"
        rel="noopener noreferrer"
        target="_blank"
      >
        Version 
        <i
          className="fas fa-code"
        />
         
      </a>
      . Built by
       
      <a
        rel="noopener noreferrer"
        target="_blank"
      />
      , copyright © 
      2017–2020
       under the
       
      <a
        href="https://choosealicense.com/licenses/agpl-3.0/"
        rel="noopener noreferrer"
        target="_blank"
      >
        AGPL-3.0 
        <i
          className="fas fa-gavel"
        />
         license
      </a>
      .
       
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        View this project on 
        <i
          className="fab fa-github"
        />
         GitHub.
      </a>
    </p>
  </footer>
</div>
`;
